src/databricks/labs/ucx/source_code/dependency_loaders.py

```diff
@@ -50,15 +55,33 @@ def load_dependency(self, dependency: Dependency) -> SourceContainer | None:
 
 
 class LocalFileLoader(DependencyLoader):
-    # see https://github.com/databrickslabs/ucx/issues/1499
+
+    def __init__(self, syspath_provider: SysPathProvider):
+        self._syspath_provider = syspath_provider
+
     def load_dependency(self, dependency: Dependency) -> SourceContainer | None:
-        raise NotImplementedError()
+        fullpath = self.full_path(dependency.path)
+        assert fullpath is not None
+        return LocalFile(fullpath, fullpath.read_text("utf-8"), Language.PYTHON)
 
     def is_file(self, path: Path) -> bool:
```

Comment:
```suggestion
    def exists(self, path: Path) -> bool:
```
this is a better name for this method semantics